src/js/containers/account/WithLatestFy.jsx

```diff
@@ -66,6 +73,115 @@ export const useLatestAccountData = () => {
     ];
 };
 
+export const useQueryParams = (params = '') => {
+    const query = new URLSearchParams(useLocation().search);
+    if (!params) {
+        const rtrn = [];
+        [...query.keys()].forEach((key) => {
+            rtrn.push({ [key]: query.get(key) });
+        });
+        return rtrn;
+    }
+    return params.map((param) => ({ [param]: query.get(param) }));
+};
+
+/*
+    * useValidTimeBasedQueryParams
+    * this function enforces validation logic for fiscal years and fiscal periods in the URL as query params
+    * by using the latest submission periods as a reference.
+*/
+export const useValidTimeBasedQueryParams = (currentUrlFy, currentUrlPeriod = null, requiredParams = ['fy', 'period']) => {
+    const history = useHistory();
+    const search = useQueryParams();
```

Comment:
```suggestion
    const existingParams = Object.fromEntries(new URLSearchParams(useLocation().search));
```

Reply:
Im just gonna keep this in a helper cuz I use it a lot.